From bd0759765e18b1a59dab8b39b42d1ac1799c042b Mon Sep 17 00:00:00 2001
From: Adam Paquette <acpaquette@usgs.gov>
Date: Tue, 31 Mar 2026 16:37:57 -0600
Subject: [PATCH] Add custom enum formatters for spdlog

---
 include/usgscsm/Distortion.h |  4 ++++
 include/usgscsm/Utilities.h  | 11 +++++++++++
 2 files changed, 15 insertions(+)

diff --git a/include/usgscsm/Distortion.h b/include/usgscsm/Distortion.h
index d5c3c51..c5018aa 100644
--- a/include/usgscsm/Distortion.h
+++ b/include/usgscsm/Distortion.h
@@ -6,10 +6,14 @@
 #include <tuple>
 #include <vector>
 
+#include <spdlog/fmt/bundled/format.h>
+
 // This should be synched with the enum in ale/Distortion.h
 enum DistortionType { RADIAL, TRANSVERSE, KAGUYALISM, DAWNFC, LROLROCNAC, CAHVOR, 
                      LUNARORBITER, RADTAN };
 
+inline auto format_as(DistortionType distortionType) { return fmt::underlying(distortionType); }
+
 // Transverse distortion Jacobian
 void transverseDistortionJacobian(double x, double y, double *jacobian,
                                   std::vector<double> const& opticalDistCoeffs);
diff --git a/include/usgscsm/Utilities.h b/include/usgscsm/Utilities.h
index 7247560..c988485 100644
--- a/include/usgscsm/Utilities.h
+++ b/include/usgscsm/Utilities.h
@@ -16,6 +16,17 @@
 #include <Warning.h>
 #include <csm.h>
 
+#include <spdlog/fmt/bundled/format.h>
+
+namespace csm
+{
+   namespace param
+   {
+        inline auto format_as(Type type) { return fmt::underlying(type); }
+        inline auto format_as(Set set) { return fmt::underlying(set); }
+   }
+}
+
 // Compute distorted focalPlane coordinates in mm
 void computeDistortedFocalPlaneCoordinates(
     const double &line, const double &sample, const double &sampleOrigin,
-- 
2.50.1 (Apple Git-155)

